pkg/distributor/distributor_test.go

```diff
@@ -288,8 +288,18 @@ func TestDistributor_Push(t *testing.T) {
 
 			request := makeWriteRequest(tc.samples.startTimestampMs, tc.samples.num, tc.metadata, false)
 			response, err := ds[0].Push(ctx, request)
-			assert.Equal(t, tc.expectedResponse, response)
-			assert.Equal(t, tc.expectedError, err)
+
+			if tc.expectedError == nil {
+				require.NoError(t, err)
+				assert.Equal(t, emptyResponse, response)
+			} else {
+				assert.Nil(t, response)
+				assert.EqualError(t, err, tc.expectedError.Error())
+
+				// Assert that downstream GRPC statuses are passed back upstream
+				_, ok := httpgrpc.HTTPResponseFromError(err)
```

Comment:
Would it be worth it to also look at the first returned value and check if the status code is the one that we expect?

Reply:
I considered that, but it didn't seem that valuable. Since we're mocking the ingester, we'd be testing that we return the [value we pass](https://github.com/grafana/mimir/pull/3219/files#diff-16b7662ec0e247fa72bf86b63ec8558a8cea51ac2280baf05a25887578132c44R60) to the mock. Testing that we get back `http.StatusInternalServerError` didn't seem that useful, since if we broke the distributor to always return a 500, the test would still pass. We could pass in some unique code, I suppose, but even then this feels like something we should test at an integration or end-to-end level, especially since the error code conversion that we saw happened in a [different package](https://github.com/grafana/mimir/blob/main/pkg/util/push/push.go#L125-L128) entirely.

Reply:
makes sense, thanks for explaining :+1: